pkg/dcmtime/datetime_test.go

```diff
@@ -0,0 +1,1344 @@
+			BadValue: "10100203040506.456789+1",
+		},
+		{
+			Name:     "TZ_DoubleSign",
+			BadValue: "10100203040506.456789++0102",
+		},
+		{
+			Name:     "TZ_BadSign",
+			BadValue: "10100203040506.456789&0102",
+		},
+		{
+			Name:     "ExtraDigit_Milliseconds",
+			BadValue: "10100203040506.4567891+0102",
+		},
+		{
+			Name:     "ExtraDigit_Milliseconds_NoTZ",
+			BadValue: "10100203040506.4567891",
+		},
+		{
+			Name:     "ExtraDigit_Seconds",
+			BadValue: "101002030405061.456789+0102",
+		},
+		{
+			Name:     "ExtraDigit_Seconds_NoTZ",
+			BadValue: "101002030405061.456789",
+		},
+		{
+			Name:     "ExtraDigit_Seconds_NoMilliseconds",
+			BadValue: "101002030405061",
+		},
+		{
+			Name:     "MissingDigit_Seconds",
+			BadValue: "1010020304056.456789+0102",
+		},
+		{
+			Name:     "MissingDigit_Seconds_NoTZ",
+			BadValue: "1010020304056.456789",
+		},
+		{
+			Name:     "MissingDigit_Seconds_NoMilliseconds",
+			BadValue: "1010020304056",
+		},
+		{
+			Name:     "MissingDigit_Minutes",
+			BadValue: "10100203045.456789+0102",
+		},
+		{
+			Name:     "MissingDigit_Minutes_NoTZ",
+			BadValue: "10100203045.456789",
+		},
+		{
+			Name:     "MissingDigit_Minutes_NoMilliseconds",
+			BadValue: "10100203045",
+		},
+		{
+			Name:     "MissingDigit_Hours",
+			BadValue: "101002034.456789+0102",
+		},
+		{
+			Name:     "MissingDigit_Hours_NoTZ",
+			BadValue: "101002034.456789",
+		},
+		{
+			Name:     "MissingDigit_Hours_NoMilliseconds",
+			BadValue: "101002034",
+		},
+		{
+			Name:     "MissingDigit_Days",
+			BadValue: "1010023.456789+0102",
+		},
+		{
+			Name:     "MissingDigit_Days_NoTZ",
+			BadValue: "1010023.456789",
+		},
+		{
+			Name:     "MissingDigit_Days_NoMilliseconds",
+			BadValue: "1010023",
+		},
+		{
+			Name:     "MissingDigit_Months",
+			BadValue: "10102.456789+0102",
+		},
+		{
+			Name:     "MissingDigit_Months_NoTZ",
+			BadValue: "10102.456789",
+		},
+		{
+			Name:     "MissingDigit_Months_NoMilliseconds",
+			BadValue: "10102.456789",
+		},
+		{
+			Name:     "MissingDigit_Years",
+			BadValue: "101.456789+0102",
+		},
+		{
+			Name:     "MissingDigit_Years_NoTZ",
+			BadValue: "101.456789",
+		},
+		{
+			Name:     "MissingDigit_Years_NoMilliseconds",
+			BadValue: "101.456789",
+		},
+		{
+			Name:     "NoMillisecondsWithSeparator",
+			BadValue: "10100203040506.",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			_, err := ParseDatetime(tc.BadValue)
+			if !errors.Is(err, ErrParseDT) {
+				t.Errorf("expected ErrParseDT, got %v", err)
+			}
+		})
+	}
+}
+
+func TestDatetime_Methods(t *testing.T) {
+	testCases := []struct {
+		TimeVal        time.Time
+		ExpectedDCM    string
+		ExpectedString string
+		Precision      PrecisionLevel
+		NoOffset       bool
+	}{
+		// Full Precision, with offset
+		{
+			TimeVal: time.Date(
+				1010,
+				2,
+				3,
+				4,
+				5,
+				6,
+				456789000,
+				time.FixedZone(
+					"",
+					-3720,
+				),
+			),
```

Comment:
back to the line breaking/unbreaking comments. I don't think we should add newlines after every parameter to time.Date function in the tests in this PR. These parameters are unlabelled, and line breaking on short parameters (like integers) doesn't really aid readability at all in my opinion. In fact, for tests cases in particular, this makes it take longer to scan through the various test cases since the code buffer ends up much longer.

https://github.com/golang/go/wiki/CodeReviewComments#line-length

Reply:
Hah, sorry. I've been trying to watch myself on this since our last PR, but I tend to code on a narrow screen (laptop) so I do this by instinct to avoid horizontal scroll (also my IDE displays the parameter names in function calls like this so the line gets REALLY long, and splitting it up visually looks more like a readable table).

Not to mention at my last job 88 character line width was enforced with extreme zealousness regardless of language so it's a hard habit to break. 

Will change, and try to not slip up on in the future when making PRs for this repo.

Reply:
Thanks and no worries! I guess the Go developers made this convention and stick with it. I think this is because maybe those folks use editors that wrap the lines (avoiding hscroll, I believe GitHub can do this too). In this case in particular, I think the tradeoff is reasonable to not break the lines and condense the amount of space the test takes up (e.g. imo readability isn't reduced much at all, but ability to grok the code is a bit better--and we stay inline with the general line length convention).